Short-Answer/Algorithms_Answers.md

```diff
@@ -2,14 +2,53 @@
 
 ## Exercise I
 
-a)
 
+a)  a = 0
+    while (a < n * n * n):  # the loop is based on n*n*n => O(n^3)
+      a = a + n * n         # this makes loop go faster by n*n, we do multipliply first from PEMDAS
+                        # the next while loop should see a + n * n < n * n * n
+                        # dividing out the n*n results in a < n    
+                        OVERALL complexity is => O(n)
 
-b)
 
+b)  sum = 0                 # assignment O(1)
+    for i in range(n):      # loop n times  O(n)    
+      j = 1                 # assignment O(1)
+      while j < n:          # test condition depends on how fast j will increment to get to n
+        j *= 2              # assignment O(1)
+        sum += 1            # assignment O(1)
+
+                    Regardless of while loop, for loop will run n times
+                    OVERALL  complexity is => O(n)
 
```

Comment:
The outer for loop will run n times, since it is incrementing i by 1 each iteration.

However, the inner loop is doubling the value of j every time it runs. This means that the number of times the inner while loop runs is not growing at a linear rate - we would describe its runtime as O(log(n)).